tests/test_client_functional.py

```diff
@@ -2166,6 +2166,36 @@ async def handler_redirect(request):
     assert data == body
 
 
+@pytest.mark.parametrize(
+    "invalid_redirect_url",
+    (
+        # yarl.URL.__new__ raises ValueError
+        "http://:/",
+        "http://www.example.com/../filename.html",
+        "http://example.org:non_int_port/",
+        # yarl.URL.origin raises ValueError
+        "http:/",
+        "http:/example.com",
+        "http:///example.com",
+    ),
+)
+async def test_invalid_redirect_url(
+    aiohttp_client: Any, invalid_redirect_url: Any
+) -> None:
+    body = b"redirect"
+    headers = {hdrs.LOCATION: invalid_redirect_url}
+
+    async def handler_request(request):
+        return web.Response(status=301, body=body, headers=headers)
```

Comment:
Body is irrelevant, right? Can we just drop it?

Reply:
Right, it's not needed - removed.